presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java

```diff
@@ -336,6 +336,25 @@ public void testCastToVarchar()
         assertInvalidCast("cast(JSON '{ \"x\" : 123}' as VARCHAR)");
     }
 
+    @Test
+    public void testEquals()
+    {
+        assertFunction("json_parse('{ \"a\": \"1.1\" , \"b\": \"2.3\" , \"c\": { \"d\": \"314E-2\" }}') = json_parse('{ \"a\": \"1.1\" , \"b\": \"2.3\" , \"c\" : { \"d\" : \"314E-2\" }}')", BOOLEAN, true);
+    }
+
+    @Test
+    public void testNotEquals()
+    {
+        assertFunction("JSON '{ \"a\": 1 , \"b\": 2 , \"c\": { \"d\": 3 }}' != JSON '{ \"a\": 1 , \"b\": 2 , \"c\" : { \"d\" : 4 }}'", BOOLEAN, true);
+    }
+
+    @Test
+    public void testIsDistinctFrom()
```

Comment:
The existing tests are in `testJsonEquality`. I agree that they shouldn't be there. Feel free to move them in a preparing commit. And then add your test cases there.

Add a test case where only left is null, a test case where only right is null, and a test case where result is false.